File: app/src/main/java/com/persona5dex/models/room/SearchSuggestionDao.java
package com.persona5dex.models.room;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Query;
import android.database.Cursor;

/**
 * Created by Rechee on 11/27/2017.
 */

@Dao
public interface SearchSuggestionDao {
    @Query("select * from searchSuggestions where suggest_text_1 like :query\n" +
            "union\n" +
            "select searchSuggestions.* from personaShadowNames\n" +
            "inner join searchSuggestions on personaShadowNames.suggestion_id = searchSuggestions._id\n" +
            "where personaShadowNames.shadow_name like :query")
    Cursor getSuggestions(String query);
}

File: app/src/main/java/com/persona5dex/models/room/SearchSuggestionDao.kt
package com.persona5dex.models.room

import androidx.room.Dao
import androidx.room.Query
import com.persona5dex.searchSuggestions.PersonaSearchSuggestion
import com.persona5dex.searchSuggestions.SearchSuggestion

/**
 * Created by Rechee on 11/27/2017.
 */
@Dao
interface SearchSuggestionDao {
    @Query("""
        select id, name as lineOne, arcana as lineTwo, 1 as type, gameId from personas
        WHERE name LIKE :query
    """)
    fun getPersonaSearchSuggestions(query: String?): List<PersonaSearchSuggestion>

    @Query("""
        select id, name as lineOne, effect as lineTwo, 2 as type from skills
        WHERE name LIKE :query
    """)
    fun getSkillSearchSuggestions(query: String?): List<SearchSuggestion>
}